layouts/_default/baseof.html

```diff
@@ -24,6 +24,46 @@
         <link rel="stylesheet" href="{{ .RelPermalink }}">
         {{ end }}
 
+        {{ $aboutDict := dict "@type" "Organization" }}
+        {{ if .Site.Params.contactName }} 
+            {{ $aboutDict = merge (dict "name" .Site.Params.contactName) $aboutDict }}
+        {{ end }}
+        {{ if .Site.Params.contactEmail }} 
+            {{ $aboutDict = merge (dict "email" .Site.Params.contactEmail) $aboutDict }}
+        {{ end }}
+
+        {{ $sameAsSlice := slice }}
+        {{ if .Site.Params.facebookUrl }}
+            {{ $sameAsSlice = $sameAsSlice | append .Site.Params.facebookUrl }}
+        {{ end }}
+        {{ if .Site.Params.twitterUrl }}
+            {{ $sameAsSlice = $sameAsSlice | append .Site.Params.twitterUrl }}
+        {{ end }}
+        {{ if .Site.Params.linkedinUrl }}
+            {{ $sameAsSlice = $sameAsSlice | append .Site.Params.linkedinUrl }}
+        {{ end }}
+        {{ if .Site.Params.blueskyUrl }}
+            {{ $sameAsSlice = $sameAsSlice | append .Site.Params.blueskyUrl }}
+        {{ end }}
+
+        {{ $urlDict := dict }}
+        {{ if .Site.BaseURL }}
+        {{ $urlDict = dict "url" .Site.BaseURL }}
+        {{ else }}
+        {{ $urlDict = dict "url" (absURL "") }}
+        {{ end }}
+
+        {{ $schemaOrg := dict 
+            "@context" "https://schema.org" 
+            "@type" "WebSite" 
+            "about" $aboutDict
+            "sameAs" $sameAsSlice
+        }}
+        {{ $schemaOrg = merge $urlDict $schemaOrg }}
+        <script type="application/ld+json">
+        {{ $schemaOrg }}
+        </script>
+
```

Comment:
As I was having kind of a problem generating the commas correctly for the JSON file I ended up with this big but resilient solution! I was not totally sure about the .Site.BaseURL but since they do not recommend using this function in the Hugo documentation (https://gohugo.io/methods/site/baseurl/) I just put a fallback option (just in case).